File: app/src/androidTest/java/com/bbtest/tools/NetworkInfo.java
package com.bbtest.tools;

import android.content.Context;
import android.net.wifi.WifiConfiguration;
import android.net.wifi.WifiManager;

import java.lang.reflect.Method;

import static androidx.test.core.app.ApplicationProvider.getApplicationContext;

public class NetworkInfo {
    private Context context = getApplicationContext();

    public void closeWifi() {
        operateWifi(false);
    }

    public void openWifi() {
        operateWifi(true);
    }

    public void operateWifi(boolean isOpenWifi) {
        WifiManager wifiManager = (WifiManager) context.getSystemService(context.WIFI_SERVICE);
        if (isOpenWifi) {
            if (!wifiManager.isWifiEnabled()) {
                wifiManager.setWifiEnabled(true);
            }
        } else if (wifiManager.isWifiEnabled()) {
            wifiManager.setWifiEnabled(false);
        }
    }


    public boolean isWiFiEnabled(Context context) {
        WifiManager wifimanager = (WifiManager) context.getSystemService(context.WIFI_SERVICE);
        try {
            Method method = wifimanager.getClass().getDeclaredMethod("isWifiApEnabled");
            method.setAccessible(true);
            return (Boolean) method.invoke(wifimanager);
        } catch (Throwable ignored) {
        }
        return false;
    }

    public boolean configApState(Context context) {
        WifiManager wifimanager = (WifiManager) context.getSystemService(context.WIFI_SERVICE);
        WifiConfiguration wificonfiguration = null;
        try {
            // if WiFi is on, turn it off
            if (isWiFiEnabled(context)) {
                wifimanager.setWifiEnabled(false);
            }
            Method method = wifimanager.getClass().getMethod("setWifiApEnabled", WifiConfiguration.class, boolean.class);
            method.invoke(wifimanager, wificonfiguration, !isWiFiEnabled(context));
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }
}

File: app/src/androidTest/kotlin/com/bbtest/tools/NetworkInfo.kt
package com.bbtest.tools

import android.content.Context
import android.net.wifi.WifiConfiguration
import android.net.wifi.WifiManager
import androidx.test.core.app.ApplicationProvider.getApplicationContext

class NetworkInfo {
    private val context: Context = getApplicationContext()

    fun closeWifi() {
        operateWifi(false)
    }

    fun openWifi() {
        operateWifi(true)
    }

    fun operateWifi(isOpenWifi: Boolean) {
        val wifiManager = context.getSystemService(Context.WIFI_SERVICE) as WifiManager
        if (isOpenWifi) {
            if (!wifiManager.isWifiEnabled) {
                wifiManager.isWifiEnabled = true
            }
        } else if (wifiManager.isWifiEnabled) {
            wifiManager.isWifiEnabled = false
        }
    }

    fun isWiFiEnabled(context: Context): Boolean {
        val wifiManager = context.getSystemService(Context.WIFI_SERVICE) as WifiManager
        return try {
            val method = wifiManager.javaClass.getDeclaredMethod("isWifiApEnabled")
            method.isAccessible = true
            method.invoke(wifiManager) as Boolean
        } catch (_: Throwable) {
            false
        }
    }

    fun configApState(context: Context): Boolean {
        val wifiManager = context.getSystemService(Context.WIFI_SERVICE) as WifiManager
        val wifiConfiguration: WifiConfiguration? = null
        return try {
            if (isWiFiEnabled(context)) {
                wifiManager.isWifiEnabled = false
            }
            val method = wifiManager.javaClass.getMethod(
                "setWifiApEnabled",
                WifiConfiguration::class.java,
                Boolean::class.javaPrimitiveType,
            )
            method.invoke(wifiManager, wifiConfiguration, !isWiFiEnabled(context))
            true
        } catch (e: Exception) {
            e.printStackTrace()
            false
        }
    }
}
